e2e_test/mock.go

```
@@ -0,0 +1,98 @@
+	}
+
+	json.NewEncoder(w).Encode(response)
+}
+
+func (mb *MockBackend) handleDefault(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	response := map[string]string{
+		"message": "mock backend response",
+		"path":    r.URL.Path,
+	}
+	json.NewEncoder(w).Encode(response)
```

golangci-lint

Error return value of `(*encoding/json.Encoder).Encode` is not checked

(errcheck)